src/JsonSchema/Constraints/Factory.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the JsonSchema package.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace JsonSchema\Constraints;
+
+use JsonSchema\Exception\InvalidArgumentException;
+use JsonSchema\Uri\UriRetriever;
+use JsonSchema\Validator;
+
+/**
+ * Factory for centralize constraint initialization.
+ */
+class Factory
```

Comment:
I am okay with the name, but I think we would be best served to stick with the convention established in release 1.4.0: https://github.com/justinrainbow/json-schema/releases/tag/1.4.0 (PR #136)

That being said, I think the suggestion @alecsammon put forth is good guidance.